WcaOnRails/app/webpacker/lib/requests/routes.js.erb

```diff
@@ -1,3 +1,9 @@
+function jsonToQueryString(json) {
+  return Object.keys(json)
+    .filter((key) => json[key] !== undefined && json[key] !== null)
+    .map((key) => `${encodeURIComponent(key)}=${encodeURIComponent(json[key])}`).join('&');
+}
+
```

Comment:
I feel like vanilla JS `URLSearchParams` should have a solution for this. Perhaps the constructor supports directly passing a hash?

Reply:
Actually, a use-case that we want is, if the JSON is having any value as null, we shouldn't pass that value to the server. So, this code actually removes the keys which have value as null.

Reply:
Then it's fine to remove the nulls "manually" and _then_ letting vanilla JS `URLSearchParams` do the rest.

The problem is that you're doing _everything_ manually, including encoding the param names and glueing together the `?` and `=` symbols. Let `URLSearchParams` handle as much as reasonably possible, and only add the functionality on top that really isn't there.